eg-api.php

```diff
@@ -188,6 +195,11 @@
 		error_log("emailPetitions was not set");
 	}
 	error_log("Finished api request.");
+	if (validAPIKey($_REQUEST, $db)) {
+		writeToResourceLog(validAPIKey($_REQUEST, $db),"eg-api.php",$log_trail);
```

Comment:
as an aside, we run validAPIKey a number of times in this script.  Is it worth storing the userid once and then just checking the userid over and over again instead of rehitting the database each time?  You aren't caching the value anywhere.